analyzere/utils.py

```diff
@@ -92,3 +92,18 @@ def vectorize(values):
     if isinstance(values, list):
         return ','.join(str(v) for v in values)
     return values
+
+
+def vectorize_range(values):
+    """
+    This function is for url endoding.
+    Takes a value or a tuple or list of tuples and returns a single result,
+    tuples are joined by "," if necessary, elements in tuple are joined by '_'
+    """
+    if isinstance(values, tuple):
+        return '_'.join(str(i) for i in values)
+    elif isinstance(values, list):
+        if all([isinstance(item, tuple) for item in values]):
+            return ','.join('_'.join(str(i) for i in v) for v in values)
+        raise Exception('Items in the list must be tuples')
```

Comment:
More specific exception?

Reply:
Use an exception class that derives from exception - not the base exception type.

Reply:
oh I see